portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -49,11 +49,26 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery commentsPrepared = datastore.prepare(commentsQuery);
 
-    // Loop through each Comment entity and store in an ArrayList.
+    // Get maxComments query and check it is valid.
+    int maxComments = getMaxComments(request);
+    if (maxComments == -1) {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter an integer between 1 and 10.");
+      return;
+    }
+
+    // Loop through each Comment entity until all comments are seen or until the max number of
+    // comments have been reached, and store in an ArrayList.
     List<String> comments = new ArrayList<>();
+    int countComments = 0;
     for (Entity entity : commentsPrepared.asIterable()) {
       String comment = (String) entity.getProperty("comment");
       comments.add(comment);
+
+      countComments++;
+      if (countComments >= maxComments) {
+        break;
+      }
```

Comment:
Instead of keeping track of a counter and breaking, why not use a for loop up to the minimum of maxComments and the length of the iterable?

Reply:
In PR #15 I did something involving a while loop? If you still think it's not as streamlined as it could be then I can edit it.

Reply:
I think this is fine, but the code could be more concise if you used something like:
for(int countComments = 0; commentsIterator.hasNext() && countComments < maxComments; countComments++)

This is optional, though.